### PR TITLE
Assorted updates

### LINKED ISSUE #133: Update rxjs to the same version as the latest inquirer
Currently the version of rxjs used in this project is the same as inquirer 6.x. This doesn't allow flattening of deps if one is using the latest inquirer 8.x. And, unfortunately, rxjs is a huge dependency with ~3K files :/ https://packagephobia.com/result?p=rxjs%406.6.7%2Crxjs%407.5.2

Could you update to the latest version please? Or even better, see if you could drop/replace rxjs? If you manage to drop/replace it, it doesn't even need to be a major version.

Thanks!

### PR DESCRIPTION
**Pull request by XhmikosR**
CI preview: https://github.com/XhmikosR/inquirer-autocomplete-prompt/actions?query=branch%3Adev

@mokkabonna let me know if you want me to move a patch to a new PR since I'm not sure if these changes should need a major/minor bump :)

Closes #133 while at it.

**Better not squash and merge**.

PS1. would be nice if you could drop rxjs and/or any other dependencies (without reinventing the wheel ofc, so maybe only run-async?), see #133 :)
PS2. you can remove any Travis CI references in apps/integrations etc after this is merged
PS3: you can probably remove lodash completely along with `util.inherits` later
PS4: I have a major updates branch, not sure if you want to land this later since it seems to only affect devDependencies: https://github.com/XhmikosR/inquirer-autocomplete-prompt/compare/dev...dev-major

Closes #133, closes #135, closes #136 along the way

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,26 +27,24 @@
...
+    "rxjs": "^7.5.2"
+  },
   "devDependencies": {
-    "chai": "^4.2.0",
-    "coveralls": "^3.1.0",
+    "c8": "^7.11.0",
+    "chai": "^4.3.4",
     "eslint": "^6.8.0",
-    "flow-bin": "^0.123.0",
+    "flow-bin": "0.123.0",
```
Comment (XhmikosR) [inline package.json:41]: I had to pin this because the newer versions result in errors. I'll leave updating this to you later :)

Comment (XhmikosR) [reply package.json]: I managed to update flow-bin in the last patch, but I'm not familiar with flow. Let me know if this works for you :)

File: index.js [hunk 2]
```diff
@@ -213,56 +216,55 @@ class AutocompletePrompt extends Base {
...
 
     // Store this promise for check in the callback
-    self.lastPromise = thisPromise;
+    const lastPromise = thisPromise;
 
-    return thisPromise.then(function inner(choices) {
+    return thisPromise.then((choices) => {
       // If another search is triggered before the current search finishes, don't set results
-      if (thisPromise !== self.lastPromise) return;
+      if (thisPromise !== lastPromise) return;
```
Comment (XhmikosR) [inline index.js:245]: Unsure if this check makes sense; it seems `lastPromise` isn't changed anywhere else?

**Comment by mokkabonna** [comment]
Thanks a lot for the work! 

I haven't been able to properly see through all changes (i have skimmed through it though and it looks clean) and will probably merge, test, release if I find nothing to object to. Give it a week or so I imagine.

Short question, is there any intentional functional changes in this PR? Seems all technical right?

**Comment by XhmikosR** [comment]
Yeah, most changes should be safe. That being said, if you plan to make it a major version bump due to the rxjs major update, maybe it's worth including https://github.com/XhmikosR/inquirer-autocomplete-prompt/compare/dev...dev-major too?

**Comment by XhmikosR** [comment]
BTW please check my OP comments and https://github.com/mokkabonna/inquirer-autocomplete-prompt/pull/134#discussion_r791940042 for later :)

File: index.js (hunk 2 above)
Comment (mokkabonna) [reply index.js:245]: Makes perfect sense. Probably used it elsewhere before.

**Comment by mokkabonna** [comment]
Again, thanks alot! Good and clean commits, on point.

**Comment by XhmikosR** [comment]
You are welcome :)

How about https://github.com/XhmikosR/inquirer-autocomplete-prompt/compare/dev...dev-major? I think after the changes here, the next version should probably be a major one. If so, the linked patch makes sense too.

**Comment by mokkabonna** [comment]
I have merged dev-major in as well.  And removed lodash and util.inherit. 

What patch?

**Comment by XhmikosR** [comment]
All good, I just noticed you cherry picked the major deps patch too :)

>Also how do you suggest I get rid of rxjs? I use it to subscribe to event.keypress etc, is there a simple fucntion I can write that can accomplish the same?

That is something I don't know TBH, otherwise I'd have submitted a PR myself. :/

**Comment by mokkabonna** [comment]
I could write a simple if statement guard in the onkeypress/onsubmit functions, but takewhile stops the subscription alltogether as soon as the condition is false.

**Comment by XhmikosR** [comment]
If you think you can make it work that's OK, otherwise don't worry about it, at least not until inquirer removes rxjs there (since rxjs will be a dependency of inquirer already, so you only need to keep in sync to reduce the impact like I did here).

**Pull request merged**